# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/softdog/hotfixes.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./softdog.c.orig	1999-01-06 16:14:23.000000000 +0100
+++ ./softdog.c	2005-02-21 12:32:29.040661432 +0100
@@ -26,6 +26,13 @@
 #include <signal.h>
 #include <errno.h>
 #include <getopt.h>
+
+#define _LINUX_STRING_H_
+#define __KERNEL__
+#  include <asm/types.h>
+#  include <linux/kdev_t.h>
+#  include <linux/list.h>
+#undef __KERNEL__
 #include <linux/fs.h>
 
 #define VERSION "1.21"
